File: app/src/main/java/org/tasks/filters/SnoozedFilter.java
package org.tasks.filters;

import static org.tasks.data.Alarm.TYPE_SNOOZE;

import android.os.Parcel;

import com.todoroo.andlib.sql.Criterion;
import com.todoroo.andlib.sql.Join;
import com.todoroo.andlib.sql.QueryTemplate;
import com.todoroo.astrid.api.Filter;
import com.todoroo.astrid.data.Task;

import org.tasks.data.Alarm;

public class SnoozedFilter extends Filter {

  public static final Creator<SnoozedFilter> CREATOR =
      new Creator<>() {

        /** {@inheritDoc} */
        @Override
        public SnoozedFilter createFromParcel(Parcel source) {
          SnoozedFilter item = new SnoozedFilter();
          item.readFromParcel(source);
          return item;
        }

        /** {@inheritDoc} */
        @Override
        public SnoozedFilter[] newArray(int size) {
          return new SnoozedFilter[size];
        }
      };

  public SnoozedFilter(String listingTitle) {
    super(listingTitle, getQueryTemplate());
  }

  private SnoozedFilter() {}

  private static QueryTemplate getQueryTemplate() {
      return new QueryTemplate()
              .join(Join.inner(Alarm.TABLE, Task.ID.eq(Alarm.TASK)))
              .where(Criterion.and(Task.DELETION_DATE.lte(0), Alarm.TYPE.eq(TYPE_SNOOZE)));
  }

  @Override
  public boolean supportsHiddenTasks() {
    return false;
  }
}

File: app/src/main/java/org/tasks/filters/SnoozedFilter.kt
package org.tasks.filters

import org.tasks.data.sql.Criterion.Companion.and
import org.tasks.data.sql.Functions.now
import org.tasks.data.sql.Join.Companion.inner
import org.tasks.data.sql.QueryTemplate
import org.tasks.data.entity.Task
import kotlinx.parcelize.Parcelize
import org.tasks.data.entity.Alarm
import org.tasks.data.dao.TaskDao.TaskCriteria.activeAndVisible
import org.tasks.themes.TasksIcons

@Parcelize
data class SnoozedFilter(
    override val title: String,
) : Filter {
    override val icon: String
        get() = TasksIcons.SNOOZE

    override val sql: String
        get() = QueryTemplate()
            .join(inner(Alarm.TABLE, Task.ID.eq(Alarm.TASK)))
            .where(
                and(
                    activeAndVisible(),
                    Alarm.TYPE.eq(Alarm.TYPE_SNOOZE),
                    Alarm.TIME.gt(now()),
                )
            )
            .toString()

    override fun supportsHiddenTasks() = false

    override fun areItemsTheSame(other: FilterListItem): Boolean {
        return other is SnoozedFilter
    }
}
